src/filters/pixelate.ts

```diff
@@ -49,17 +59,34 @@ export function pixelate(image: Image, options: PixelateOptions): Image {
       for (let row = 0; row < image.height; row += cellSize) {
         const currentCellWidth = Math.min(cellSize, image.width - column);
         const currentCellHeight = Math.min(cellSize, image.height - row);
-
-        const center = getCenter({
+        let valuesOfSector = getArrayOfValues(image, channel, {
```

Comment:
To calculate mean you do not need to create a temporary array. It still takes time and memory.

You can directly create 3 (internal) methods:

* getCellCenter()
* getCellMedian()
* getCellMean()